### PR TITLE
Kms 24816 moodle   changes to versions

### PR DESCRIPTION
**Pull request by noychn-kaltura**

### CODE REVIEW

File: local/kaltura/version.php [hunk 1]
```diff
@@ -25,8 +25,45 @@
...
+$kalturaPluginVersion = "";
+if ($localKalturaPluginVersionRecords) {
+    foreach ($localKalturaPluginVersionRecords as $key => $localKalturaPluginVersionRecord) {
+        $kalturaPluginVersion = $localKalturaPluginVersionRecord->value;
+        break;
+    }
+}
+
+if ($kalturaPluginVersion == 20210620311) {
+    $pluginsRecords = $DB->get_records_select('config_plugins', "name = 'version' AND value = '$kalturaPluginVersion'");
```
Comment (noychn-kaltura) [inline local/kaltura/version.php]: I tried to put this code in a function but I got a white screen in the browser, when removed the function it was fixed

**Review by alon1980 — APPROVED** [review]

File: local/kaltura/version.php [hunk 2]
```diff
@@ -25,8 +25,45 @@
...
+
+global $DB;
+
+$localKalturaPluginVersionRecords = $DB->get_records_select('config_plugins', "plugin = 'local_kaltura' AND name = 'version'");
+
+$kalturaPluginVersion = "";
+if ($localKalturaPluginVersionRecords) {
+    foreach ($localKalturaPluginVersionRecords as $key => $localKalturaPluginVersionRecord) {
+        $kalturaPluginVersion = $localKalturaPluginVersionRecord->value;
+        break;
```
Comment (gonenradai) [inline local/kaltura/version.php]: this indicates you either expect only one or handle only the first and don't care about the others.
if that's the case, why foreach loop?

File: local/kaltura/version.php [hunk 3]
```diff
@@ -25,8 +25,45 @@
...
+if ($kalturaPluginVersion == 20210620311) {
+    $pluginsRecords = $DB->get_records_select('config_plugins', "name = 'version' AND value = '$kalturaPluginVersion'");
+
+    foreach ($pluginsRecords as $record) {
+        $record->value = 2021051700;
+        $DB->update_record('config_plugins', $record);
+    }
+}
+else if ($kalturaPluginVersion == 20201215310 || $kalturaPluginVersion == 20210620310) {
+    $pluginsRecords = $DB->get_records_select('config_plugins', "name = 'version' AND value = '$kalturaPluginVersion'");
```
Comment (gonenradai) [inline local/kaltura/version.php]: these blocks seem identical except for the target record value being set.
you can probably do an if/elseif/elseif block to only set a variable ```$resetDbVersion``` to the desired value, and then write once the line to get records and the loop to update records.

**Review by gonenradai — COMMENTED** [review]
see my comments

**Review by gonenradai — APPROVED** [review]

**Pull request merged**